### PR TITLE
Jump defination

### PR DESCRIPTION
**Pull request by RachelBra**

### CODE REVIEW

File: library/math/numbersLine/src/components/Jump.tsx [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+interface IProps {
+  element: IElement;
+  idClick: number;
+}
+
+const Jump = ({ idClick, element }: IProps) => {
+  const { dragElements, setDragElements } = useNumbersLineContext();
+  const targetRef = React.useRef<any>(null);
+  const chandeHidenumbers = () => {
+    let newelements = dragElements.map((e: IElement) => (e.id === idClick ? { ...e, hideNumber: !e.hideNumber } : e));
```
Comment (sarasla) [inline library/math/numbersLine/src/components/Jump.tsx]: item instead e

File: library/math/numbersLine/src/components/Jump.tsx [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    setDragElements(newelements);
+  };
+  return (
+    <>
+      <div ref={targetRef} className="absolute top-[35%] left-[50%] w-[134px] h-[124px]">
+        <img src={jumpArrowPlus} alt="Menu Arroa" />
+        <div className={baseJumpClassName} onClick={() => chandeHidenumbers()}>
```
Comment (sarasla) [inline library/math/numbersLine/src/components/Jump.tsx]: typo changeHidenumbers

File: library/math/numbersLine/src/components/toolbar/ActionIcons.tsx [hunk 3]
```diff
@@ -2,35 +2,55 @@ import { useEffect, useRef, useState } from "react";
...
 }
-const IconsToolbar = ({ iconUrl }: IProps) => {
+const IconsToolbar = ({ type, iconUrl }: IProps) => {
   const [isHovered, setIsHovered] = useState(false);
   const [isOpen, setIsOpen] = useState(false);
-  const [type, setType] = useState("");
 
   const wrapperRef = useRef<HTMLDivElement>(null);
 
+  useEffect(() => {
```
Comment (sarasla) [inline library/math/numbersLine/src/components/toolbar/ActionIcons.tsx]: remove log

File: library/math/numbersLine/src/context/numbersLineContext.tsx [hunk 4]
```diff
@@ -26,26 +25,22 @@ export const NumbersLineContext = React.createContext({
...
-  const { saveData, getData } = useLocalStorage();
 
-  useEffect(() => {
-    setDragElement(getData("element"));
-  }, []);
 
   useEffect(() => {
-    dragElement.length > 0 && saveData("element", dragElement);
-  }, [dragElement]);
+    console.log("dragElements: ", dragElements);
```
Comment (sarasla) [inline library/math/numbersLine/src/context/numbersLineContext.tsx]: remove log

File: library/math/numbersLine/src/lib/utils.ts [hunk 5]
```diff
@@ -7,9 +7,11 @@ export function cn(...inputs: ClassValue[]) {
 }
 
 export const calculationWidthScreen = (windowWidth: number)=>{
-  return -windowWidth*((LineRange.hundred-RulerLenth.hundred)/RulerLenth.hundred)
+  return -1*windowWidth*((LineRange.hundred-RulerLenth.hundred)/RulerLenth.hundred)
```
Comment (sarasla) [inline library/math/numbersLine/src/lib/utils.ts]: why -1 and not -

**Review by sarasla — CHANGES_REQUESTED** [review]

File: library/math/numbersLine/src/lib/utils.ts (hunk 5 above)
Comment (RachelBra) [reply library/math/numbersLine/src/lib/utils.ts]: i think it is more clear 
but i change if you think it is not

Comment (sarasla) [reply library/math/numbersLine/src/lib/utils.ts]: ok.

File: library/math/numbersLine/src/components/Jump.tsx [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+  const { dragElements, setDragElements } = useNumbersLineContext();
+  const targetRef = React.useRef<any>(null);
+  const changeHidenumbers = () => {
+    let newelements = dragElements.map((item: IElement) => (item.id === idClick ? { ...item, hideNumber: !item.hideNumber } : item));
+    setDragElements(newelements);
+  };
+  return (
+    <>
+      <div ref={targetRef} className="absolute top-[35%] left-[50%] w-[134px] h-[124px]">
+        <img src={jumpArrowPlus} alt="Menu Arroa" />
```
Comment (sarasla) [inline library/math/numbersLine/src/components/Jump.tsx:23]: arrow

File: library/math/numbersLine/src/components/Toolbar.tsx [hunk 7]
```diff
@@ -15,12 +16,12 @@ const Toolbar = () => {
         <img className="p-5" src={fullscreen} alt="Fullscreen Toolbar" />
       </div>
       <div className="flex ml-[4.25rem]">
-        <IconsToolbar iconUrl={back} />
-        <IconsToolbar iconUrl={icons} />
-        <IconsToolbar iconUrl={eye} />
-        <IconsToolbar iconUrl={writing} />
-        <IconsToolbar iconUrl={text} />
-        <IconsToolbar iconUrl={jump} />
+        <IconsToolbar type={TypeActionIconsToolbar.raload} iconUrl={back} />
```
Comment (sarasla) [inline library/math/numbersLine/src/components/Toolbar.tsx]: maby use const with type and url and then loop on this var.

**Review by sarasla — CHANGES_REQUESTED** [review]

**Review by sarasla — APPROVED** [review]

**Pull request merged**